tests/apps/test_clique.py

```diff
@@ -252,6 +349,14 @@ def test_bad_node_select(self, dim):
         with pytest.raises(ValueError, match="Node selection method not recognized"):
             clique.swap(s, graph, node_select="")
 
+    def test_bad_weights(self, dim, graph):

```

Comment:
Is this function different from the one at line 221? Might be better to use different names?

Reply:
It is the same test applied to different functions. I steer towards preferring them to have the same name as doing so allows you to pool them together using the `-k "test_bad_weights"` argument in `pytest`.